Tests/iaas/entropy/entropy-check.py

```diff
@@ -0,0 +1,377 @@
+        wrong = [f"{key}={value}" for key, value in attributes.items() if extra_specs.get(key) != value]
+        miss_opt = [key for key in optional if extra_specs.get(key) is None]
+        if wrong:
+            message = f"Flavor '{flavor.name}' missing recommended attributes: {', '.join(wrong)}"
+            # only report missing optional attributes if recommended are missing as well
+            # reasoning here is that these optional attributes are merely a hint for implementers
+            # and if the recommended attributes are present, we assume that implementers have done their job already
+            if miss_opt:
+                message += f"; additionally, missing optional attributes: {', '.join(miss_opt)}"
+            logger.info(message)
+
+
+def install_test_requirements(fconn):
+    # the following commands seem to be necessary for CentOS 8, but let's not go there
+    # because, frankly, that image is ancient
+    # sudo sed -i -e "s|mirrorlist=|#mirrorlist=|g" /etc/yum.repos.d/CentOS-*
+    # sudo sed -i -e "s|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g" /etc/yum.repos.d/CentOS-*
+    # Try those commands first that have a high chance of success (Ubuntu seems very common)
+    commands = (
+        # use ; instead of && after update because an error in update is not fatal
+        # also, on newer systems, it seems we need to install rng-tools5...
+        ('aptitude', 'apt-get -v && (sudo apt-get update ; sudo apt-get install -y rng-tools5 || sudo apt-get install -y rng-tools)'),
+        ('dnf', 'sudo dnf install -y rng-tools'),
+        ('yum', 'sudo yum -y install rng-tools'),
+        ('pacman', 'sudo pacman -Syu rng-tools'),
+    )
+    for name, cmd in commands:
+        try:
+            fconn.run(cmd, hide=True)
+        except invoke.exceptions.UnexpectedExit as e:
+            logger.debug(f"Error running '{name}': {e.result.stderr.strip()}")
+        else:
+            return
+    logger.debug("No package manager worked; proceeding anyway as rng-utils might be present nonetheless.")
+
+
+def check_vm_requirements(fconn, image_name):
+    try:
+        entropy_avail = fconn.run('cat /proc/sys/kernel/random/entropy_avail', hide=True).stdout.strip()
+        if entropy_avail != "256":
+            logger.error(
+                f"VM '{image_name}' didn't have a fixed amount of entropy available. "
+                f"Expected 256, got {entropy_avail}."
+            )
+
+        install_test_requirements(fconn)
+        fips_data = fconn.run('cat /dev/random | rngtest -c 1000', hide=True, warn=True).stderr
+        failure_re = re.search(r'failures:\s\d+', fips_data, flags=re.MULTILINE)
+        if failure_re:
+            fips_failures = failure_re.string[failure_re.regs[0][0]:failure_re.regs[0][1]].split(" ")[1]
+            if int(fips_failures) > 3:
+                logger.error(
+                    f"VM '{image_name}' didn't pass the FIPS 140-2 testing. "
```

Comment:
I guess a specific fips 140-2 test should at least link to a document/the standard describing what should be tested.

Reply:
The script just does what the corresponding standard requires:

https://github.com/SovereignCloudStack/standards/blob/main/Standards/scs-0101-v1-entropy.md#compute-nodes